src/datascience-ui/react-common/postOffice.ts

```diff
@@ -74,38 +84,70 @@ export class PostOffice implements IDisposable {
     }
 
     public acquireApi(): IVsCodeApi | undefined {
```

Comment:
Seems like this could be split up more? Or generalized into two different functions instead of multiple if points? I know some of it might be duped in each function, but I believe there's two distinct paths here?

- postKerneMessage = native case
- acquireVsCodeApi = webview case

Reply:
I'm suggesting:

- acquireApi (highest level)
- acquireNativeApi (for native case when new function - isNative - returns true). This function would do the work for bnding for the native case.
- acquireWebViewApi - This function would do the code we used to have?

Reply:
Were you thinking like just break this function up into the kernel / vs code specific functions? Or more like factor our a full common interface. Full interface feels kinda heavy, but it might read better.

Reply:
I think two different functions would be good enough and make it clearer which path you were on.

Reply:
I was actually starting down the path of an interface / class refactor. What do you think about this? @rchiodo